src/runtime/virtcontainers/virtiofsd.go

```diff
@@ -147,7 +147,7 @@ func (v *virtiofsd) Start(ctx context.Context, onQuit onQuitFunc) (int, error) {
 	go func() {
 		scanner := bufio.NewScanner(stderr)
 		for scanner.Scan() {
-			v.Logger().WithField("source", "virtiofsd").Info(scanner.Text())
```

Comment:
@tiezhuoyu Can you explain if `scanner.Text()` will be called if what you said `scanner readis nothing from viriofsd stderr` is right.

Reply:
`scanner.Text()` is not excuted, because `scanner.Scan()` blocks to read from buffer. I think it is enough with only `scanner.Scan()`.

Reply:
Naive question : if the scanners basically sits reading nothing from the pipe, why is the scanner needed at all in the first place ?

Reply:
Yes, I miss this... Both `scanner.Scan` and `cmd.Process.Wait()` do the same thing, to block untile virtiofsd exits. Since read nothing from stderr, it is not necessary to create a `scanner` and call `scanner.Scan()`.

Reply:
> Yes, I miss this... Both `scanner.Scan` and `cmd.Process.Wait()` do the same thing, to block until virtiofsd exits. Since read nothing from stderr, it is not necessary to create a `scanner` and call `scanner.Scan()`.

That was exactly my point. :smile: